### PR TITLE
fix(recurrent cache): admit exact checkpoint hits during active decode

### PR DESCRIPTION
**Pull request by voipmonitor**
## Behavior and contract

Admit an exact recurrent checkpoint hit even while other requests are decoding.
The restored hidden state needs one isolated logits-only step; resident decodes
remain allocated and resume together with the admitted request on the following
step. Running-first scheduling otherwise makes the cached request wait until
all resident decodes finish.

Reserve that step only for a known complete checkpoint at the waiting queue
head. Capacity limits and admission pauses retain their existing behavior.
If allocation or admission cannot progress, schedule resident requests instead
of leaving the device idle. Partial checkpoint hits use the ordinary path.

The public boundary-state test also covers request slots zero and one, including
selector state, saved hidden states, draft state and partial attention pages.
The scalar-safe restore implementation comes from #674 by @logprobz; it is not
duplicated here. Merge #674 as well to run the slot-one integration case.

## Validation

Status: **qualified** for the scheduler invariant and composed serving cases
below, not a general language-model quality certification.

- The minimal scheduler test fails without the admission change. The fixed
  suite passes 199 scheduler, fairness and micro-slicing cases, including full
  capacity, allocation failure and admission pause controls.
- Seven boundary/input-state GPU tests pass with scalar-safe restoration.
- In a composed Qwen3.8-Flash-Next-NVFP4 MTP3 server, cold and exact-checkpoint
  batches at C1, C2, C3, C4, C8 and C16 return the required JSON on TP1 and TP2.
  Scheduler gauges reach each requested concurrency. All requests use
  temperature 1, top-p .95 and top-k 20. V2 FULL_AND_PIECEWISE remains enabled.
- A 32,768-token prefill followed by C3/C4/C8/C16 cold and restored batches
  also passes. These serving checks include the separate padded-QSA fix in
  #658; they are composition results, not an isolated speedup claim.

Reproduce in a built CUDA vLLM environment:

```bash
uv run .venv/bin/python -m pytest -q \
  tests/v1/core/test_scheduler.py \
  tests/v1/core/test_prefill_compute_share_scheduler.py \
  tests/v1/core/test_micro_slicing.py
uv run .venv/bin/python -m pytest -q \
  tests/v1/worker/test_gpu_input_batch_v2.py -k boundary
```

## Scope and review

Targets `dev/jovian-judgement`. #664 controls prefill compute sharing and keeps
logits-only steps isolated; it does not admit a waiting full checkpoint while
running requests continually consume the scheduling step. Searches in this fork
and upstream found no equivalent cached-waiter admission fix. #674 supplies the
independent scalar-restore fix. #682 adds GLM DCP/DFlash restore geometry.

AI assistance was used for implementation, inspection and tests. Human
maintainer review is required before merge. The assembly and evidence are
tracked in #662.

### CODE REVIEW

**Comment by voipmonitor** [comment]
@coderabbitai review
